geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/DensityIterator.scala

```diff
@@ -294,5 +219,12 @@ object DensityIterator extends Logging {
     }
     table
   }
+}
 
+case class DensityIteratorResult(geometry: Geometry,
+                                 densityGrid: SparseMatrix = HashBasedTable.create[Double, Double, Long]()) extends Result {
+  override def addToFeature(featureBuilder: SimpleFeatureBuilder): Unit = {
+    featureBuilder.add(DensityIterator.encodeSparseMatrix(densityGrid))
+    featureBuilder.add(geometry)
```

Comment:
so, this is the way the DensityIterator was working before -- returning the geometry from the last feature, I am not sure if this is an essential part of its behavior or not.  Can anyone who knows weigh in? (@jnh5y?)